### ISSUE TITLE
ruboto setup problem

### ISSUE THREAD

**Issue by KCErb** [ruboto/ruboto#734]
Hi all, when I run `ruboto setup -y` I get the following

```
WARN: Unresolved specs during Gem::Specification.reset:
      rake (~> 10.0)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
Java runtime             : Found
Java Compiler            : Found
Apache ANT               : Found
Android Package Installer: Found
Android Emulator         : Found
F, [2015-06-08T20:55:57.192620 #4762] FATAL -- : undefined method `[]' for nil:NilClass (NoMethodError)
/Users/KC/.rvm/gems/ruby-2.1.2/gems/ruboto-1.3.0/lib/ruboto/util/setup.rb:205:in `check_for_haxm'
```
I tried playing around with the `check_for_haxm` method like so

```ruby
@haxm_installer_loc = Dir[File.join(android_package_directory, 'extras', 'intel', 'Hardware_Accelerated_Execution_Manager', 'IntelHAXM*.dmg')].first
puts @haxm_installer_loc
exit
```

I get this:

```
/Users/KC/android-sdk-macosx/extras/intel/Hardware_Accelerated_Execution_Manager/IntelHAXM.dmg
```

So I *do* have the HAXM dmg, but no version number to scrape. Should I just use it as is and fix the HAXM checker to look for this as well?

**Comment by donv** [comment]
Hi @KCErb !

Thanks for the report!  Could you try master and see if this has already been fixed?

https://github.com/ruboto/ruboto#from-source

**Comment by daneb** [comment]
Is this not the same problem I resolved? Sorry coming at the tail end here and finding it hard to keep track. @KCErb are you having this problem on latest Yosemite? That was my test platform and it had proven successful.

**Comment by KCErb** [comment]
Hi @daneb, yes it is resolved on master! Sorry for the confusion, I looked at the gem-copy of `setup.rb` and thought it matched your PR, but I guess I was wrong because I just pulled master and tried again and everything was fine. :)

With one exception: when I tried running `bin/ruboto setup -y` it still was trying to use the gem-installed copy instead of master. In another project I work on, we correct this by requiring `bundler/setup` [like this](https://github.com/shoes/shoes4/blob/master/shoes-core/lib/shoes/ui/picker.rb#L19-L26).

So on my copy of ruboto I added a guard like this to `bin/ruboto` (I'd be happy to submit a PR if you're interested) 

```ruby
if File.exist?("Gemfile")
  # Only need bundler/setup to get our paths right--we don't need to
  # actually require the gems, since we'll do that later
  require 'bundler/setup'
end
```

With this in place I ran again:

```
KC:ruboto$ bin/ruboto setup -y
WARN: Unresolved specs during Gem::Specification.reset:
      rake (~> 10.0)
      rubyzip (~> 1.0)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
/Users/KC/.rvm/gems/ruby-2.1.2/gems/bundler-1.9.1/lib/bundler/runtime.rb:34:in `block in setup': You have already activated ruboto 1.3.0, but your Gemfile requires ruboto 1.3.1.dev. Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
	from /Users/KC/.rvm/gems/ruby-2.1.2/gems/bundler-1.9.1/lib/bundler/runtime.rb:19:in `setup'
	from /Users/KC/.rvm/gems/ruby-2.1.2/gems/bundler-1.9.1/lib/bundler.rb:122:in `setup'
	from /Users/KC/.rvm/gems/ruby-2.1.2/gems/bundler-1.9.1/lib/bundler/setup.rb:8:in `<top (required)>'
	from /Users/KC/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:135:in `require'
	from /Users/KC/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
	from /Users/KC/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:144:in `require'
	from bin/ruboto:13:in `<main>'
```

So I did a `bundle exec` and everything went swimmingly except ruboto doesn't like my HAXM:

```
KC:ruboto$ bundle exec bin/ruboto setup -y
Java runtime             : Found
Java Compiler            : Found
Apache ANT               : Found
Android Package Installer: Found
Android Emulator         : Found
Intel HAXM               : Old   1.1.2/
Android SDK Command adb  : Found
Android SDK Command dx   : Found
Platform SDK android-16  : Found

    *** Ruboto setup is OK! ***
```

In short, the problem is fixed and I'll close this out.
In long:
  * I'm curious if I can help out cleaning up how we run from source / learn how you guys do it.
  * I'm curious if I should do something about that old HAXM, since you're on Yosemite maybe you can offer some insight.
